api.ts

```diff
@@ -1656,6 +1670,8 @@ export function makeSyncTable<
     ...definition
   } = maybeRewriteConnectionForFormula(formula, connectionRequirement);
 
+  const wrappedAutocomplete = autocomplete ? makeMetadataFormula(autocomplete as MetadataFunction) : undefined;
```

Comment:
I'm not sure `makeMetadataFormula` will return the right thing as-is. I think it'll return a formula with different parameters than your new interface. So probably need a new wrapper analogous to makeMetadataFormula but using correct params. Probably implies it would be best to update the typing for autocompleteCell to a new variant of MetadataFormula as well.

Reply:
Done